gsdk/tests/rpc.rs

```diff
@@ -355,6 +359,52 @@ async fn test_program_counters() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn test_calculate_reply_for_handle() -> Result<()> {
+    let node = dev_node();
+
+    let salt = vec![];
+    let pid = ProgramId::generate_from_user(CodeId::generate(demo_ping::WASM_BINARY), &salt);
+
+    // 1. upload program.
+    let signer = Api::new(Some(&node_uri(&node)))
+        .await?
+        .signer("//Alice", None)?;
+
+    signer
+        .calls
+        .upload_program(
+            demo_ping::WASM_BINARY.to_vec(),
+            salt,
+            vec![],
+            100_000_000_000,
+            0,
+        )
+        .await?;
+
+    assert!(
+        signer.api().gprog(pid).await.is_ok(),
+        "Program not exists on chain."
+    );
+
+    // 2. calculate reply for handle
+    let reply_info = signer
+        .rpc
+        .calculate_reply_for_handle(None, pid, b"PING".to_vec(), 100_000_000_000, 0, None)
+        .await?;
+
+    assert_eq!(
+        reply_info,
+        ReplyInfo {
+            payload: b"PONG".to_vec(),
+            value: 0,
+            code: ReplyCode::Success(SuccessReplyReason::Manual)
+        }
+    );
```

Comment:
I would assert Alice's account balance hasn't changed, i.e. it was for free.

Reply:
I guess it's unnecessary since there's no even signing for the call

Reply:
Same for "immutability of the call", but lets keep it cz already coded